Feature/add geographic reporting (#40)
fivetran/dbt_microsoft_ads_source: CHANGELOG.md

# dbt_microsoft_ads_source v0.12.0

## Schema Changes
**3 total changes • 0 possible breaking changes
| Table/Column                                      | Change Type | Old Name | New Name                                  | Notes                                                             |
|---------------------------------------------------|-------------|----------|-------------------------------------------|-------------------------------------------------------------------|
| stg_microsoft_ads__geographic_daily_report_tmp       | New Model   |     |  | Temp model added for `geographic_performance_daily_report`.               |
| stg_microsoft_ads__geographic_daily_report          | New Model   |          |    | Staging model added for `geographic_performance_daily_report`.         |

> | budget, budget_id, budget_type, language           | New Columns   |          |    | New columns added to the stg_microsoft_ads__campaign_history model.         |

Reviewer: Let's hold off on adjusting this till we align as a team on Monday, but I was thinking [this](https://github.com/fivetran/dbt_facebook_ads/blob/feature-update/add-geo-reports/CHANGELOG.md) could be a good format for new columns maybe? Or I'll adopt your format 🤔

Author: Isn't this the same format? Something looks weird with the formatting as well.
<img width="999" alt="Screenshot 2025-04-25 at 4 16 46 PM" src="https://github.com/user-attachments/assets/96bc7e63-1063-43e5-95dd-374a6a0b7d3a" />

> | budget, budget_id, budget_type, language           | New Columns   |          |    | New columns added to the `stg_microsoft_ads__campaign_history` model.         |

Reviewer: Let's largely go with your New Columns format but switch it a little to align with this (from [slack](https://fivetran.slack.com/archives/C02FG7499UK/p1745868115067119?thread_ts=1745341464.307869&cid=C02FG7499UK))
![image](https://github.com/user-attachments/assets/a8274e4c-d6b4-4805-972c-eaabf58329c2)

So basically put the table name in the first column (with a DAG/docs link) and add the columns to the `New Name` column

Author: Some modifications based on the recent Slack feedback, but updated!

## Feature Updates
- Added the `geographic_performance_daily_report` source table and downstream staging models. See above for schema change details and new models added.

>   - For dbt Core users: This table is disabled by default. If you do not sync this table or would like to enable these new models, you can enable the models by setting the  `microsoft_ads__using_geographic_daily_report` variable to `true` in your `dbt_project.yml` file (`false` by default).

Reviewer: could be good to include a link to the README section

Author: Added.

- Included the `microsoft_ads__geographic_passthrough_metrics` passthrough variable in the above mentioned new staging models. Refer to the [README](https://github.com/fivetran/dbt_microsoft_ads_source?tab=readme-ov-file#passing-through-additional-metrics) for more details.

> - Introduced the above mentioned new columns to the `stg_microsoft_ads__campaign_history` model.

Reviewer: We could probably remove this line

Author: Agreed.

>   - For dbt Core users: This table is disabled by default. If you do not sync this table or would like to enable these new models, you can enable the models by setting the  `microsoft_ads__using_geographic_daily_report` variable to `true` in your `dbt_project.yml` file (`false` by default). Refer to the [README](https://github.com/fivetran/dbt_microsoft_ads_source?tab=readme-ov-file#enable-geographic-reports) for more details. 

Reviewer: ```suggestion
  - For dbt Core users: This table is disabled by default. If you would like to enable these new models, you can do so by setting the  `microsoft_ads__using_geographic_daily_report` variable to `true` in your `dbt_project.yml` file (`false` by default). Refer to the [README](https://github.com/fivetran/dbt_microsoft_ads_source?tab=readme-ov-file#enable-geographic-reports) for more details. 
```

Author: Updated.
